numba/core/typed_passes.py

```diff
@@ -648,6 +651,42 @@ def _do_work_call(self, state, work_list, block, i, expr, inline_worker):
             work_list, is_method, inline_worker,
         )
 
+    def _do_work_ops(self, state, work_list, block, i, expr, inline_worker):
+
+        # search for this binop overloads with this "inline"
+        sig = state.type_annotation.calltypes[expr]
+        if sig is None:
+            return False
+
+        arg_typs = sig.args
+        templates = state.typingctx._functions.get(expr.fn, [])
```

Comment:
This should probably be something like `state.typingctx.resolve_value_type(expr.fn)` which will resolve to the `types.Function` for the `expr.fn`. The templates are in the `.templates` attr on the resolved function, so something like https://github.com/numba/numba/pull/6027/files#diff-936f29e49d280cc5a6e7b5405222612dR617 should then work.

Reply:
@stuartarchibald Yes, thanks for pointing the right way. I changed it to using `resolve_value_type`, but I didn't quite understand the next part regarding rating/sorting the candidates like in `resolve_overload`. It was so that the first template, matching arg types was selected anyway. Also it turned out that the list of templates obtained with `state.typingctx.resolve_value_type(expr.fn).templates` is larger than the one with `state.typingctx._functions.get(expr.fn, [])`.   It contains certain entries looking like not yet evaluated (they have empty `_get_template_info`), but it works since they are filtered with `inline_type = getattr(template, '_inline', None)`.